core/app/models/concerns/spree/user_address_book.rb

```diff
@@ -10,57 +10,76 @@ def find_first_by_address_values(address_attrs)
+      has_one :bill_address, through: :default_user_bill_address, source: :address
 
-      has_one :default_user_address, ->{ default }, class_name: 'Spree::UserAddress', foreign_key: 'user_id'
-      has_one :default_address, through: :default_user_address, source: :address
-      alias_method :ship_address, :default_address
+      has_one :default_user_ship_address, ->{ default_shipping }, class_name: 'Spree::UserAddress', foreign_key: 'user_id'
+      has_one :ship_address, through: :default_user_ship_address, source: :address
     end
 
-    def bill_address=(address)
-      # stow a copy in our address book too
-      address = save_in_address_book(address.attributes) if address
-      super(address)
+    def default_address
+      Spree::Deprecation.warn "This association is deprecated. Please start using #ship_address."
+      ship_address
     end
 
-    def bill_address_attributes=(attributes)
-      self.bill_address = Spree::Address.immutable_merge(bill_address, attributes)
+    def default_user_address
+      Spree::Deprecation.warn "This association is deprecated. Please start using #default_user_ship_address."
+      default_user_ship_address
     end
 
     def default_address=(address)
+      Spree::Deprecation.warn "This setter does not take into account Spree::Config.automatic_default_address and is deprecated. "\
+        "Please start using #ship_address="
       save_in_address_book(address.attributes, true) if address
     end
 
     def default_address_attributes=(attributes)
       # see "Nested Attributes Examples" section of http://apidock.com/rails/ActionView/Helpers/FormHelper/fields_for
       # this #{fieldname}_attributes= method works with fields_for in the views
       # even without declaring accepts_nested_attributes_for
-      self.default_address = Spree::Address.immutable_merge(default_address, attributes)
-    end
+      Spree::Deprecation.warn "This setter is deprecated. Please start using #ship_address_attributes="
 
-    alias_method :ship_address_attributes=, :default_address_attributes=
+      self.default_address = Spree::Address.immutable_merge(ship_address, attributes)
+    end
 
     # saves address in address book
     # sets address to the default if automatic_default_address is set to true
     # if address is nil, does nothing and returns nil
     def ship_address=(address)
-      be_default = Spree::Config.automatic_default_address
-      save_in_address_book(address.attributes, be_default) if address
+      if address
+        save_in_address_book(address.attributes,
+                             Spree::Config.automatic_default_address)
+      end
+    end
+
+    def ship_address_attributes=(attributes)
+      self.ship_address = Spree::Address.immutable_merge(ship_address, attributes)
+    end
+
+    def bill_address=(address)
+      # stow a copy in our address book too
```

Comment:
Can we remove this comment, please? I think the code is already self-explanatory

Reply:
Done